### PR TITLE
Add domain_lookup_start functionality

### LINKED ISSUE #21259: PerformanceResourceTiming: domainLookupStart
In components/net/http_loader.rs::http_fetch we need to set domain_lookup_start (if we're using a persistent connection). Otherwise, domain lookup should be performed by hyper, which is probably out of the scope of this issue.

On getting, the domainLookupStart attribute MUST return as follows:

1. The same value as fetchStart, if a persistent connection [RFC7230] is used or the resource is retrieved from relevant application caches or local resources.
2. The time immediately after the user agent before the domain data retrieval from the domain information cache, if the user agent has the domain information in cache.
3. The time immediately before the user agent starts the domain name lookup for the resource, if the last non-redirected fetch of the resource passes the timing allow check algorithm.
4. zero, otherwise.

Spec: https://w3c.github.io/resource-timing/#dom-performanceresourcetiming-domainlookupstart

### PR DESCRIPTION
**Pull request by KaczuH**
Added the domain_lookup_start functionality in http_loader.rs (http_redirect_fetch function)

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #21259 (GitHub issue number if applicable)

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because ___

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23310)

### CODE REVIEW

File: components/net/http_loader.rs [hunk 1]
```diff
@@ -675,6 +675,12 @@ pub fn http_redirect_fetch(
         .unwrap()
         .set_attribute(ResourceAttribute::FetchStart);
 
+    context
+        .timing
+        .lock()
+        .unwrap()
+        .set_attribute(ResourceAttribute::DomainLookupStart);
```
Comment (jdm) [inline components/net/http_loader.rs]: We should move this to [right before](https://github.com/servo/servo/blob/834d3193a2ed016093b6b997d29df5a607715956/components/net/http_loader.rs#L383) the HTTP request is initiated instead.

**Review by jdm — CHANGES_REQUESTED** [review]

File: components/net/http_loader.rs (hunk 1 above)
Comment (KaczuH) [reply components/net/http_loader.rs]: Thanks for quick feedback!
I've moved the code as requested.

**Comment by jdm** [comment]
Looks good! Could you squash the commits into one single commit, too?

**Comment by KaczuH** [comment]
Ok, commits are sqashed.
I'm not sure why AppVeyor build was cancelled.

**Comment by CYBAI** [comment]
Let's fix the `FAIL` one and label the latter one as `PASS`! 😄

```
{
    "status": "FAIL", 
    "group": "default", 
    "message": "assert_equals: wrong typeof object expected \"object\" but got \"undefined\"", 
    "stack": "IdlInterface.prototype.test_interface_of/<@http://web-platform.test:8000/resources/idlharness.js:2758:17\nTest.prototype.step@http://web-platform.test:8000/resources/testharness.js:1587:20\ntest@http://web-platform.test:8000/resources/testharness.js:544:21\nself.subsetTestByKey@http://web-platform.test:8000/resources/idlharness.js:54:23\nIdlInterface.prototype.test_interface_of@http://web-platform.test:8000/resources/idlharness.js:2755:13\nIdlInterface.prototype.test_object@http://web-platform.test:8000/resources/idlharness.js:2634:9\nIdlArray.prototype.test/<@http://web-platform.test:8000/resources/idlharness.js:840:17\nIdlArray.prototype.test@http://web-platform.test:8000/resources/idlharness.js:835:13\nidl_test/</<@http://web-platform.test:8000/resources/idlharness.js:3252:21\n", 
    "subtest": "PerformanceResourceTiming interface: resource must inherit property \"domainLookupStart\" with the proper type", 
    "test": "/resource-timing/idlharness.any.worker.html", 
    "line": 290996, 
    "action": "test_result", 
    "expected": "PASS"
}
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "domainLookupStart should be 0 in cross-origin request.", 
    "test": "/resource-timing/resource_TAO_zero.htm", 
    "line": 344577, 
    "action": "test_result", 
    "expected": "FAIL"
}
```

File: components/script/dom/performanceresourcetiming.rs [hunk 2]
```diff
@@ -86,8 +86,8 @@ impl PerformanceResourceTiming {
             redirect_start: 0.,
             redirect_end: 0.,
             fetch_start: fetch_start,
-            domain_lookup_start: 0.,
             domain_lookup_end: 0.,
+            domain_lookup_start: 0.,
```
Comment (jdm) [inline components/script/dom/performanceresourcetiming.rs]: We need to adjust the `from_resource_timing` method below this to initialize the value to the stored one as well.

**Review by jdm — CHANGES_REQUESTED** [review]

**Comment by jdm** [comment]
Sorry about the delay in reviewing this!

File: components/script/dom/performanceresourcetiming.rs (hunk 2 above)
Comment (KaczuH) [reply components/script/dom/performanceresourcetiming.rs]: Ok, done :)

**Comment by jdm** [comment]
That test result will need to be adjusted.

**Comment by jdm** [comment]
@KaczuH Are you planning to update the expected test result?

**Comment by KaczuH** [comment]
@jdm yes, I will. I just need day or two.
Thank you for the patience

**Comment by KaczuH** [comment]
@jdm I'm not sure if i know how to fix the failing [test](https://build.servo.org/builders/linux-rel-wpt/builds/12363).
I've removed it from `idlharness.any.js.ini` similarly as others did in related PRs.

File: tests/wpt/metadata/resource-timing/idlharness.any.js.ini [hunk 3]
```diff
@@ -82,9 +76,6 @@
   [PerformanceResourceTiming interface: resource must inherit property "transferSize" with the proper type]
     expected: FAIL
 
-  [PerformanceResourceTiming interface: resource must inherit property "domainLookupStart" with the proper type]
-    expected: FAIL
```
Comment (jdm) [inline tests/wpt/metadata/resource-timing/idlharness.any.js.ini]: Restore this particular failure (just these two lines) to this file and the test should stop reporting an unexpected failure.

Comment (KaczuH) [reply tests/wpt/metadata/resource-timing/idlharness.any.js.ini]: Thank you. Done

**Comment by jdm** [comment]
Go ahead and squash the commits into one and then this can merge :)

**Comment by KaczuH** [comment]
@jdm squash done

**Comment by jdm** [comment]
@bors-servo r+
Thanks!

**Comment by CYBAI** [comment]
```
{
    "status": "PASS", 
    "group": "default", 
    "message": null, 
    "stack": null, 
    "subtest": "PerformanceResourceTiming interface: resource must inherit property \"domainLookupStart\" with the proper type", 
    "test": "/resource-timing/idlharness.any.worker.html", 
    "line": 139027, 
    "action": "test_result", 
    "expected": "FAIL"
}
```

**Pull request merged**